### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by qtomlinson**
Add initial integration testing for harvest and definition comparison. More test cases will be added in the future to cover all the component types.

A sample run can be found at https://github.com/qtomlinson/service/actions/runs/7836818205/job/21385360669.

### CODE REVIEW

**Review by lumaxis — COMMENTED** [review]
It's great to have these as tools! But I wonder if the best place is this repo? Since these tools cover the larger pipeline and more than just the service, maybe it would be better to keep them somewhere separately? Especially since I don't see any dependency on the existing code in this repo 🤔

File: integration/tools/fetch.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+function buildPostOpts(json) {
+  return {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(json)
+  }
+}
+
+async function callFetch(url, fetchOpts) {
+  console.log(url, fetchOpts)
```
[the remark this answers is not in the document]
Comment (qtomlinson) [reply integration/tools/fetch.js]: Agreed and fixed.

File: integration/tools/fetch.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+const fetch = (...args) => import('node-fetch').then(({ default: fetch }) => fetch(...args))
```
[the remark this answers is not in the document]
Comment (qtomlinson) [reply integration/tools/fetch.js:1]: Good point!

**Comment by qtomlinson** [comment]
@lumaxis @elrayle Once reviewed, I will move this to operations repo.

File: integration/test/testConfig.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+const devApiBaseUrl = 'https://dev-api.clearlydefined.io'
+const prodApiBaseUrl = 'https://api.clearlydefined.io'
+
+const pollingInterval = 1000 * 60 * 5 // 5 minutes
+const pollingMaxTime = 1000 * 60 * 30 // 30 minutes
+
+const harvestToolVersions = [
+  ['licensee', '9.14.0'],
+  ['scancode', '30.3.0'],
+  ['reuse', '3.2.1']
```
Comment (qtomlinson) [inline integration/test/testConfig.js:10]: @lumaxis @yashkohli88, after scan tool update, the versions here will need to be updated.

**Review by lumaxis — APPROVED** [review]

File: integration/tools/fetch.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (lumaxis) [reply integration/tools/fetch.js:1]: Fwiw, if we have this in the operations repo, might even be possible to use native fetch in Node 21 right away but definitely not a big deal 😄

**Comment by qtomlinson** [comment]
Moved the PR to operations: https://github.com/clearlydefined/operations/pull/58

**Pull request closed without merging**